### PR TITLE
Add support for DHCP relay

### LINKED ISSUE #148: I cannot provision DHCP-Relay on an Network-Interface for a UDM-PRO
Hello,

I have following Part imported from an UDM-PRO:

  dhcp_dns = [
    "8.8.8.8",
    "8.8.4.4",
  ]
  dhcp_enabled        = false
  dhcp_lease          = 86400
  dhcp_start          = "192.168.160.6"
  dhcp_stop           = "192.168.160.254"
  dhcpd_boot_enabled  = false
  domain_name         = "zuhause.local"
  igmp_snooping       = false
  ipv6_interface_type = "none"
  ipv6_ra_enable      = false
  name                = "USW"
  network_group       = "LAN"
  purpose             = "corporate"
  site                = "default"
  subnet              = "192.168.160.0/24"
  vlan_id             = 33
  wan_dns             = []
  wan_egress_qos      = 0
}

Bild 18 06 21 um 17 11

Bild 18 06 21 um 17 13

If I provision the terraform part, then the dhcp-relay is disabled

Can you fix this?

Thank Uwe Blais

### PR DESCRIPTION
**Pull request by paultyng**
Fixes #148

~This is probably only supported on v6, so may have to tune the testing.~

### CODE REVIEW

File: internal/provider/resource_network_test.go [hunk 1]
```diff
@@ -287,6 +287,33 @@ func TestAccNetwork_importByName(t *testing.T) {
...
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { preCheck(t) },
+		ProviderFactories: providerFactories,
+		// TODO: CheckDestroy: ,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccNetworkConfigDHCPRelay(name, vlanID, true),
+				Check:  resource.ComposeTestCheckFunc(
+				// TODO:
```
Comment (joshuaspence) [inline internal/provider/resource_network_test.go:302]: Incomplete?

**Review by joshuaspence — APPROVED** [review]

File: internal/provider/resource_network_test.go (hunk 1 above)
Comment (paultyng) [reply internal/provider/resource_network_test.go:302]: Yeah not quite done, was just lazy about testing so I pushed it up to get it to run 😄

File: internal/provider/resource_network.go [hunk 2]
```diff
@@ -267,6 +274,12 @@ func resourceNetworkGetResourceData(d *schema.ResourceData) (*unifi.Network, err
 		return nil, fmt.Errorf("unable to convert wan_dns to string slice: %w", err)
 	}
 
+	dhcpRelay := d.Get("dhcp_relay_enabled").(bool)
+	if v := c.ControllerVersion(); dhcpRelay && !v.GreaterThanOrEqual(controllerV6) {
+		// unsure exactly what version it was added in release notes, but looks like only v6 versions
+		return nil, fmt.Errorf("dhcp_relay_enabled is only supported on controller versions >= 6.0.0")
```
Comment (paultyng) [inline internal/provider/resource_network.go:280]: This is the kind of thing I'd like to avoid with the v5 removal you know.

**Comment by joshuaspence** [comment]
Just let me know when it's ready for review 😀

**Comment by paultyng** [comment]
@joshuaspence should be good now!

**Comment by paultyng** [comment]
@joshuaspence going to go ahead and merge this, but let me know if you see any issues and I can correct in a follow-up.

**Pull request merged**